### PR TITLE
add deprecation config

### PR DESCRIPTION
**Pull request by pongad**

### CODE REVIEW

File: src/test/java/com/google/api/codegen/testdata/library_gapic.yaml [hunk 1]
```diff
@@ -504,6 +504,7 @@ interfaces:
       - go
       - java
       visibility: PACKAGE
+      deprecated: TRUE
```
Comment (garrettjonesgoogle) [inline src/test/java/com/google/api/codegen/testdata/library_gapic.yaml]: I would prefer a more flexible enumeration. Maybe:

`lifecycle: DEPRECATED`

That allows us to do something like this too:

`lifecycle: EXPERIMENTAL`

This also implies that the SurfaceNamer for each language needs to produce that language's string for each concept.

**Comment by pongad** [comment]
@garrettjonesgoogle PTAL

File: src/main/java/com/google/api/codegen/viewmodel/StaticLangApiMethodView.java [hunk 2]
```diff
@@ -89,6 +89,8 @@ public boolean isStreaming() {
   @Nullable
   public abstract LongRunningOperationDetailView operationMethod();
 
+  public abstract String releaseAnnotation();
```
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/viewmodel/StaticLangApiMethodView.java]: I  think `releaseLevelAnnotation` would be clearer.

Comment (pongad) [inline src/main/java/com/google/api/codegen/viewmodel/StaticLangApiMethodView.java]: Done

**Comment by garrettjonesgoogle** [comment]
LGTM

**Pull request merged**